prometheus/src/main/scala/filodb/prometheus/ast/Functions.scala

```diff
@@ -84,10 +85,11 @@ trait Functions extends Base with Operators with Vectors {
       val miscellaneousFunctionIdOpt = MiscellaneousFunctionId.withNameInsensitiveOption(name)
       val scalarFunctionIdOpt = ScalarFunctionId.withNameInsensitiveOption(name)
       val sortFunctionIdOpt = SortFunctionId.withNameInsensitiveOption(name)
+      val absentFunctionIdOpt = AbsentFunctionId.withNameInsensitiveOption(name)
       // Get parameters other than  series like label names. Parameters can be quoted so remove special characters
       val stringParam = allParams.filter(!_.equals(seriesParam)).
-                        filter(_.isInstanceOf[InstantExpression]).
-                        map(_.asInstanceOf[InstantExpression].realMetricName.replaceAll("^\"|\"$", ""))
+        filter(_.isInstanceOf[InstantExpression]).
```

Comment:
Instead of this, use a collect, much cleaner:

    collect {
      case e: InstantExpression => e.realMetricName.replaceAll(.....)
    }

Reply:
Also I think indentation is better when lined up against allParams